controllers/ci.controller.js

```diff
@@ -0,0 +1,159 @@
+const path = require('path');
+
+const CiModel = require("../models/CiModel");
+const { validate } = require('./common.controller');
+
+const Constants = require('../lib/constants');
+const logger = require('../lib/logger').API;
+
+// adding an application
+const ciAdd = async (req, res, next) => {
+    logger.addContext(Constants.FILE_NAME, path.basename(__filename));
+    logger.info('The ci add controller is started');
+
+    try {
+        // validation
+        const validateResult = validate(req);
+        if (!validateResult.success) {
+            return res.status(validateResult.status).json({success: false, errors: validateResult.errors});
+        }
+
+        // read request parameter data
+        const {ci_url} = req.body
+        // if ci_url already exists, return error or else create a new apps
+        const ci = await CiModel.findOne({ci_url});
+        logger.info(ci);
+        if (ci) {
+            logger.warn(`the ci_url already exists, ${ci.ci_url}`);
+            return res.status(400).json({success: false, errors: {errormessage:'Ci already exists',errorcode:'400'}});
```

Comment:
please change errormessage to message, errorcode to code.
to think create some common methods to generate response.